### PR TITLE
deprecate(discussion): Deprecates the elgg:discussion library

### LINKED ISSUE #8760: master: container breadcrumb doesn't show in discussion/add
Instead it errors because $group is undefined.

### PR DESCRIPTION
**Pull request by mrclay**
Moves the function into start.php

Fixes #8760

### CODE REVIEW

**Comment by ewinslow** [comment]
Is there a current usage/import of the library that should be removed?

File: mod/discussions/views/default/resources/discussion/add.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 $title = elgg_echo('discussion:addtopic');
 
-elgg_push_breadcrumb($group->name, "discussion/owner/$group->guid");
+elgg_push_breadcrumb($container->getDisplayName(), "discussion/owner/{$container->guid}");
```
Comment (ewinslow) [inline mod/discussions/views/default/resources/discussion/add.php]: This apparently fixes #8760 ?

File: mod/discussions/start.php [hunk 2]
```diff
@@ -82,8 +82,6 @@ function discussion_init() {
  */
 function discussion_page_handler($page) {
 
-	elgg_load_library('elgg:discussion');
```
Comment (mrclay) [inline mod/discussions/start.php]: This was the only place it was used.

File: mod/discussions/views/default/resources/discussion/add.php (hunk 1 above)
Comment (mrclay) [inline mod/discussions/views/default/resources/discussion/add.php]: Yes, I just added that to the commit message.

**Pull request merged**